### PR TITLE
integrating the Xcache Consistency Check tool into the cms-xcache image (software-4026)

### PR DESCRIPTION
**Pull request by ddavila0**

### CODE REVIEW

File: cms-xcache/image-config.d/30-save-environment.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+# This saves all the env vars passes to the ocntainer at launch in a way
+# that other non-root processes can see them
+cat /proc/1/environ | sed 's/\x00/\n/g' > /etc/environment
```
Comment (efajardo) [inline cms-xcache/image-config.d/30-save-environment.sh]: Why is this needed? This seems pretty dangerous. I would feel more comfortably only passing it the env variables that start with XC.

Comment (efajardo) [reply cms-xcache/image-config.d/30-save-environment.sh:4]: @ddavila0   :top:

Comment (ddavila0) [reply cms-xcache/image-config.d/30-save-environment.sh]: fixed

File: cms-xcache/xcache-consistency-check-wrapper.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+ENABLE=`cat /etc/environment |  grep XC_XCACHE_CONSISTENCY_CHECK | awk -F "=" '{print $2}'`
```
Comment (brianhlin) [inline cms-xcache/xcache-consistency-check-wrapper.sh:2]:
```suggestion
ENABLE=`awk -F "=" '/^XC_XCACHE_CONSISTENCY_CHECK/ {print $2}' /etc/environment`
```
This assumes that each line in the environment file is `VAR=VAL`

File: cms-xcache/xcache-consistency-check-wrapper.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+echo "ENABLE: "$ENABLE
+
+if [ "$ENABLE" == "1" ]
+then
+    echo "Starting"
```
Comment (brianhlin) [inline cms-xcache/xcache-consistency-check-wrapper.sh:7]:
```suggestion
    echo "Starting XCache consistency check..."
```

File: cms-xcache/xcache-consistency-check-wrapper.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+ENABLE=`cat /etc/environment |  grep XC_XCACHE_CONSISTENCY_CHECK | awk -F "=" '{print $2}'`
+echo "ENABLE: "$ENABLE
```
Comment (brianhlin) [inline cms-xcache/xcache-consistency-check-wrapper.sh:3]:
```suggestion
```

File: cms-xcache/xcache-consistency-check-wrapper.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+if [ "$ENABLE" == "1" ]
+then
+    echo "Starting"
+    export PYTHONPATH=/usr/lib/xcache-consistency-check/usr/lib/python2.7/site-packages/:/usr/lib/xcache-consistency-check/usr/lib64/python2.7/site-packages/
+    /usr/bin/xcache-consistency-check --config /etc/xrootd/xcache-consistency-check.cfg
+fi
```
Comment (brianhlin) [inline cms-xcache/xcache-consistency-check-wrapper.sh:10]:
```suggestion
else
    echo "XCache consistency check disabled via the environment (XC_XCACHE_CONSISTENCY_CHECK=$ENABLE)..."
fi
```

File: cms-xcache/Dockerfile [hunk 6]
```diff
@@ -4,10 +4,13 @@ LABEL maintainer OSG Software <help@opensciencegrid.org>
 RUN yum install -y cms-xcache --enablerepo=osg-testing && \
     yum clean all --enablerepo=* && rm -rf /var/cache/
 
+RUN yum install -y  xcache-consistency-check --enablerepo=osg-testing
+
```
Comment (brianhlin) [inline cms-xcache/Dockerfile:8]:
```suggestion
```

File: cms-xcache/Dockerfile [hunk 7]
```diff
@@ -4,10 +4,13 @@ LABEL maintainer OSG Software <help@opensciencegrid.org>
 RUN yum install -y cms-xcache --enablerepo=osg-testing && \
```
Comment (brianhlin) [inline cms-xcache/Dockerfile:4]:
```suggestion
RUN yum install -y --enablerepo=osg-testing \
                cms-xcache \
                xcache-consistency-check && \
```

Comment (ddavila0) [reply cms-xcache/Dockerfile]: For some reason It doesn't allow me to commit this change anymore, I guess because the file has changed with the other change suggestion in this same file. Anyways I've pushed a commit with the intended change

File: cms-xcache/xcache-consistency-check-wrapper.sh (hunk 3 above)
Comment (ddavila0) [reply cms-xcache/xcache-consistency-check-wrapper.sh]: I cannot longer commit this suggestion so I've pushed a new commit for this changed
